### ISSUE TITLE
LoggerSettings__c cannot see the org config in tests, settings like SystemLogMessageFormat__c revert to defaults

### ISSUE THREAD

**Issue by arbokrad** [jongpie/NebulaLogger#210]
Since LoggerSettings__c is a custom setting, it is not automatically available with any pre-configured org defaults without using the `@IsTest(SeeAllData=true)` annotation.

```
@IsTest
static void Nebula_Harness() {
	// outputs using the in-code default format (creates its own LoggerSettings__c)
	Logger.debug( 'Pre-LoggerSettings Update' );

	LoggerSettings__c settings = Logger.getUserSettings();
	settings.SystemLogMessageFormat__c = '{OriginLocation__c} -- {Message__c}';

	// outputs using the updated format (above)
	Logger.debug( 'Post-LoggerSettings Update' );
}

@IsTest
static void Nebula_WithDefined() {
	LoggerSettings__c orgDefaults = LoggerSettings__c.getOrgDefaults();
	orgDefaults.SystemLogMessageFormat__c = '{OriginLocation__c} -- {Message__c}';
	upsert orgDefaults;

	// outputs using the updated format (above)
	Logger.debug( 'Logging with custom system message format set in org default settings' );
}
```

The result is a different logger format for real debug logs vs. apex test debug logs.

Workarounds include creating this in every test class (in a `@TestSetup` method, maybe?)... or just living with the default log format.

**Comment by jongpie** [comment]
@arbokrad thanks for logging this! Given this issue, I think it makes sense to replace the field `LoggerSettings__c.SystemLogMessageFormat__c` with a custom metadata record so that test classes can have the same consistent format. I have some other cleanup work I'd like to do anyway for some of the custom metadata types, so I'll include the change for `SystemLogMessageFormat__c` in that work. Tentatively, my plan is:

- v4.6.8 release - PR #191 - this PR is unrelated to this issue, but I want to wrap it up before moving on to something else
- v4.6.9 release - Replace `LoggerSettings__c.SystemLogMessageFormat__c` with a custom metadata record, and make some other custom metadata type changes

**Comment by jongpie** [comment]
@arbokrad I've released [v4.6.9](https://github.com/jongpie/NebulaLogger/releases/tag/v4.7.9) last night, which includes changes for this - the formatting is now controlled via a custom metadata type, `LoggerParameter__mdt` - whatever format you use should now apply to your Apex tests too. Let me know if you run into any more issues, or need anything else!
